ext/phar/phar_object.c

```diff
@@ -2755,18 +2762,21 @@ PHP_METHOD(Phar, setAlias)
 		zend_throw_exception_ex(phar_ce_PharException, 0, "phar \"%s\" is persistent, unable to copy on write", phar_obj->archive->fname);
 		RETURN_THROWS();
 	}
-	if (phar_obj->archive->alias_len && NULL != (fd_ptr = zend_hash_str_find_ptr(&(PHAR_G(phar_alias_map)), phar_obj->archive->alias, phar_obj->archive->alias_len))) {
-		zend_hash_str_del(&(PHAR_G(phar_alias_map)), phar_obj->archive->alias, phar_obj->archive->alias_len);
+	if (phar_obj->archive->alias && NULL != (fd_ptr = zend_hash_find_ptr(&(PHAR_G(phar_alias_map)), phar_obj->archive->alias))) {
+		zend_hash_del(&(PHAR_G(phar_alias_map)), phar_obj->archive->alias);
 		readd = 1;
 	}
```

Comment:
Can't we just do `zend_hash_del` here and check for `SUCCESS/FAILURE` to know if an entry existed?

Reply:
No, we need fd_ptr

Reply:
Oh actually nvm we don't need it actually, strange code...